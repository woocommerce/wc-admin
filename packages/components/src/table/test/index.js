/**
 * External dependencies
 */
import { render, screen } from '@testing-library/react';
import { toHaveClass } from '@testing-library/jest-dom/matchers';

/**
 * Internal dependencies
 */
import TableCard from '../index';
import mockHeaders from './data/table-mock-headers';
import mockData from './data/table-mock-data';

expect.extend( { toHaveClass } );

describe( 'TableCard', () => {
	it( 'should render placeholder table while loading', () => {
		render(
			<TableCard
				title="Revenue"
				headers={ mockHeaders }
				isLoading={ true }
				rows={ [] }
				rowsPerPage={ 5 }
				totalRows={ 5 }
			/>
		);

		expect( screen.getByRole( 'group', { hidden: true } ) ).toHaveClass(
			'is-loading'
		);
	} );

	it( 'should not render placeholder table when not loading', () => {
		render(
			<TableCard
				title="Revenue"
				headers={ mockHeaders }
				isLoading={ false }
				rows={ mockData }
				rowsPerPage={ 5 }
				totalRows={ 5 }
			/>
		);

		expect( screen.getByRole( 'group' ) ).not.toHaveClass( 'is-loading' );
	} );
} );
