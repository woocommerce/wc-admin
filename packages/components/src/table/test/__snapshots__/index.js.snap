// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCard should render placeholders for Table and TableSummary while loading 1`] = `
<div>
  <div
    class="components-card is-size-medium woocommerce-table has-menu css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-medium e1q7k77g1 css-jsa6v9-Flex-HeaderUI eboqfv50"
    >
      <h2
        class="css-1ahfdc3-Text e15wbhsk0"
        size="20"
      >
        Revenue
      </h2>
      <div
        class="woocommerce-table__actions"
      />
      <div
        class="woocommerce-ellipsis-menu"
      >
        <div
          class="components-dropdown"
        >
          <button
            aria-expanded="false"
            class="components-button woocommerce-ellipsis-menu__toggle"
            title="Choose which values to display"
            type="button"
          >
            <svg
              class="gridicon gridicons-ellipsis"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M7 12c0 1.104-.896 2-2 2s-2-.896-2-2 .896-2 2-2 2 .896 2 2zm12-2c-1.104 0-2 .896-2 2s.896 2 2 2 2-.896 2-2-.896-2-2-2zm-7 0c-1.104 0-2 .896-2 2s.896 2 2 2 2-.896 2-2-.896-2-2-2z"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="components-card__body css-xmjzce-BodyUI e1q7k77g3"
    >
      <span
        class="screen-reader-text"
      >
        Your requested data is loading
      </span>
      <div
        aria-hidden="true"
        aria-labelledby="caption-0"
        class="woocommerce-table__table is-loading"
        role="group"
      >
        <table>
          <caption
            class="woocommerce-table__caption screen-reader-text"
            id="caption-0"
          >
            Revenue
          </caption>
          <tbody>
            <tr>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Date
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Orders
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Total Sales
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Refunds
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Coupons
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Taxes
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Shipping
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Net Sales
                </span>
              </th>
            </tr>
            <tr>
              <th
                class="woocommerce-table__item"
                scope="row"
              >
                <span
                  class="is-placeholder"
                />
              </th>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
            </tr>
            <tr>
              <th
                class="woocommerce-table__item"
                scope="row"
              >
                <span
                  class="is-placeholder"
                />
              </th>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
            </tr>
            <tr>
              <th
                class="woocommerce-table__item"
                scope="row"
              >
                <span
                  class="is-placeholder"
                />
              </th>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
            </tr>
            <tr>
              <th
                class="woocommerce-table__item"
                scope="row"
              >
                <span
                  class="is-placeholder"
                />
              </th>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
            </tr>
            <tr>
              <th
                class="woocommerce-table__item"
                scope="row"
              >
                <span
                  class="is-placeholder"
                />
              </th>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
              <td
                class="woocommerce-table__item"
              >
                <span
                  class="is-placeholder"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="components-flex components-card__footer is-size-medium e1q7k77g4 css-5hda2c-Flex-FooterUI eboqfv50"
    >
      <ul
        class="woocommerce-table__summary is-loading"
        role="complementary"
      >
        <li
          class="woocommerce-table__summary-item"
        >
          <span
            class="is-placeholder"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`TableCard should render table along with summary data when row and summary data is present 1`] = `
<div>
  <div
    class="components-card is-size-medium woocommerce-table has-menu css-1xs3c37-CardUI e1q7k77g0"
  >
    <div
      class="components-flex components-card__header is-size-medium e1q7k77g1 css-jsa6v9-Flex-HeaderUI eboqfv50"
    >
      <h2
        class="css-1ahfdc3-Text e15wbhsk0"
        size="20"
      >
        Revenue
      </h2>
      <div
        class="woocommerce-table__actions"
      />
      <div
        class="woocommerce-ellipsis-menu"
      >
        <div
          class="components-dropdown"
        >
          <button
            aria-expanded="false"
            class="components-button woocommerce-ellipsis-menu__toggle"
            title="Choose which values to display"
            type="button"
          >
            <svg
              class="gridicon gridicons-ellipsis"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M7 12c0 1.104-.896 2-2 2s-2-.896-2-2 .896-2 2-2 2 .896 2 2zm12-2c-1.104 0-2 .896-2 2s.896 2 2 2 2-.896 2-2-.896-2-2-2zm-7 0c-1.104 0-2 .896-2 2s.896 2 2 2 2-.896 2-2-.896-2-2-2z"
                />
              </g>
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="components-card__body css-xmjzce-BodyUI e1q7k77g3"
    >
      <div
        aria-hidden="false"
        aria-labelledby="caption-1"
        class="woocommerce-table__table"
        role="group"
      >
        <table>
          <caption
            class="woocommerce-table__caption screen-reader-text"
            id="caption-1"
          >
            Revenue
          </caption>
          <tbody>
            <tr>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Date
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Orders
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Total Sales
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Refunds
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Coupons
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Taxes
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Shipping
                </span>
              </th>
              <th
                class="woocommerce-table__header"
                role="columnheader"
                scope="col"
              >
                <span
                  aria-hidden="false"
                >
                  Net Sales
                </span>
              </th>
            </tr>
            <tr>
              <th
                class="woocommerce-table__item"
                scope="row"
              >
                10/01/2018
              </th>
              <td
                class="woocommerce-table__item"
              >
                1
              </td>
              <td
                class="woocommerce-table__item"
              >
                €411.00
              </td>
              <td
                class="woocommerce-table__item"
              >
                €0.00
              </td>
              <td
                class="woocommerce-table__item"
              >
                €10.00
              </td>
              <td
                class="woocommerce-table__item"
              >
                €0.00
              </td>
              <td
                class="woocommerce-table__item"
              >
                €0.00
              </td>
              <td
                class="woocommerce-table__item"
              >
                €411.00
              </td>
            </tr>
            <tr>
              <th
                class="woocommerce-table__item"
                scope="row"
              >
                10/02/2018
              </th>
              <td
                class="woocommerce-table__item"
              />
              <td
                class="woocommerce-table__item"
              >
                €0.00
              </td>
              <td
                class="woocommerce-table__item"
              >
                €0.00
              </td>
              <td
                class="woocommerce-table__item"
              >
                €0.00
              </td>
              <td
                class="woocommerce-table__item"
              >
                €0.00
              </td>
              <td
                class="woocommerce-table__item"
              >
                €0.00
              </td>
              <td
                class="woocommerce-table__item"
              >
                €0.00
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="components-flex components-card__footer is-size-medium e1q7k77g4 css-5hda2c-Flex-FooterUI eboqfv50"
    >
      <ul
        class="woocommerce-table__summary"
        role="complementary"
      >
        <li
          class="woocommerce-table__summary-item"
        >
          <span
            class="woocommerce-table__summary-value"
          >
            100
          </span>
          <span
            class="woocommerce-table__summary-label"
          >
            orders
          </span>
        </li>
        <li
          class="woocommerce-table__summary-item"
        >
          <span
            class="woocommerce-table__summary-value"
          >
            20
          </span>
          <span
            class="woocommerce-table__summary-label"
          >
            customers
          </span>
        </li>
        <li
          class="woocommerce-table__summary-item"
        >
          <span
            class="woocommerce-table__summary-value"
          >
            50
          </span>
          <span
            class="woocommerce-table__summary-label"
          >
            products
          </span>
        </li>
        <li
          class="woocommerce-table__summary-item"
        >
          <span
            class="woocommerce-table__summary-value"
          >
            0
          </span>
          <span
            class="woocommerce-table__summary-label"
          >
            coupons
          </span>
        </li>
        <li
          class="woocommerce-table__summary-item"
        >
          <span
            class="woocommerce-table__summary-value"
          >
            $12345.67
          </span>
          <span
            class="woocommerce-table__summary-label"
          >
            net sales
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
